src/main/java/com/epam/lowcost/db/DBinitialize.java

```diff
@@ -3,77 +3,72 @@
 
 import org.apache.commons.dbcp2.BasicDataSource;
 
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
 
 public class DBinitialize {
 
-    private BasicDataSource basicDataSource;
     private Connection conn;
     private Statement stm;
 
     public DBinitialize(BasicDataSource basicDataSource) throws SQLException {
-        this.basicDataSource = basicDataSource;
+
         conn = basicDataSource.getConnection();
         stm = conn.createStatement();
-        createTable();
-        populateTable();
-        selectFromTable();
-
-    }
-
-
-
-    public void createTable() throws SQLException {
-
-        String sql =  "CREATE TABLE   REGISTRATION " +
-                "(id INTEGER not NULL, " +
-                " first VARCHAR(255), " +
-                " last VARCHAR(255), " +
-                " age INTEGER, " +
-                " PRIMARY KEY ( id ))";
-
-        stm.executeUpdate(sql);
-
+        initiateDB();
+        printAllDb();
     }
-    public void populateTable() throws SQLException{
-        String sql = "INSERT INTO Registration " + "VALUES (100, 'Zara', 'Ali', 18)";
-
-        stm.executeUpdate(sql);
-        sql = "INSERT INTO Registration " + "VALUES (101, 'Mahnaz', 'Fatma', 25)";
 
-        stm.executeUpdate(sql);
-        sql = "INSERT INTO Registration " + "VALUES (102, 'Zaid', 'Khan', 30)";
 
-        stm.executeUpdate(sql);
-        sql = "INSERT INTO Registration " + "VALUES(103, 'Sumit', 'Mittal', 28)";
+    private void initiateDB() throws SQLException {
 
-        stm.executeUpdate(sql);
-        System.out.println("Inserted records into the table...");
+        File fl = new File(this.getClass().getResource("/createTable").getFile());
+        try (BufferedReader br = Files.newBufferedReader(fl.toPath())) {
 
+            String sql = br.readLine();
+            while (sql != null) {
+                stm.addBatch(sql);
+                sql = br.readLine();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        stm.executeBatch();
```

Comment:
I didn't find where you use close connection and statement.
```
Connection conn = null;
   Statement stmt = null;
   try{
      conn = DriverManager.getConnection(DB_URL, USER, PASS);
      stmt = conn.createStatement();
      String sql = "CREATE DATABASE STUDENTS";
      stmt.executeUpdate(sql);
   }catch(SQLException se){
      //Handle errors for JDBC
      se.printStackTrace();
   }catch(Exception e){
      //Handle errors for Class.forName
      e.printStackTrace();
   }finally{
      //finally block used to close resources
      try{
         if(stmt!=null)
            stmt.close();
      }catch(SQLException se2){
      }// nothing we can do
      try{
         if(conn!=null)
            conn.close();
      }catch(SQLException se){
         se.printStackTrace();
      }//end finally try
   }//end try
```
Something like this

Reply:
Changed DBinitialize constructor to close Connection and Statement.